welly/las.py

```diff
@@ -0,0 +1,366 @@
+# the LASFile curve header metadata
+curve_item = CurveItem().__dict__
+
+# set pandas precision higher to not automatically round the curve data
+pd.set_option('precision', 10)
+
+
+def from_las(file_ref, **kwargs):
+    """
+    Read LAS file with lasio and parse every LAS dataset to 2 pd.DataFrames:
+        1. curve data
+        2. header metadata
+
+    Only LAS 1.2 and 2.0 are currently supported. LAS 3.0 will be supported when lasio LAS 3.0 work is finished:
+    https://github.com/kinverarity1/lasio/issues/5. The design of this reader already accommodates for LAS 3.0
+    functionality where 1 lasio.LASFile can contain multiple 1D, 2D or 3D dataset entries. Also see lasio documentation:
+    https://lasio.readthedocs.io/en/latest/header-section.html#tutorial
+
+    Args:
+        file_ref (file-like object, str): either a filename, an open file
+            object, or a string containing the contents of a LAS file.
+        **kwargs: The additional keyword arguments are propagated to the lasio reader so you can use when reading in a
+                  LAS file. Find the routines of the different keyword possibilities look here:
+                    * :func:`lasio.reader.open_with_codecs` - manage issues relate to character encodings
+                    * :meth:`lasio.LASFile.read` - control how NULL values and errors are handled during parsing
+
+    Returns:
+        datasets (Dict['dataset_name': (data (pd.DataFrame), header (pd.DataFrame))]):
+            A dict that has an item for every dataset found in the LAS file. Any LAS dataset has a header and data part
+            that is mapped 1-to-1 to two separate pd.DataFrames and put into a tuple. See below for an example of how
+            that is structured.
+
+
+    Example of how a return would be structured:
+
+    datasets = {
+        'Curves':   (data, header), # for LAS 1.2 & LAS 2.0
+        'ASCII':    (data, header), # for LAS 3.0
+        'Drilling': (data, header), # for LAS 3.0
+        'Core[1]':  (data, header), # for LAS 3.0 - Run 1
+        'Core[2]':  (data, header)  # for LAS 3.0 - Run 2
+    }
+
+    Where:
+        data (pd.DataFrame):   where:
+                                - every row represents a data index.
+                                - every column represents a data variable. Column name is the data variable mnemonic.
+        header (pd.DataFrame): where:
+                                - every row represents a line read from LAS file and columns.
+                                - column 1-5 - directly parsed from the HeaderItem.__dict__:
+                                    'original_mnemonic' - original mmnemonic
+                                    'mnemonic' - mnemonic
+                                    'unit' - unit
+                                    'value' - value
+                                    'descr' - description
+                                - column 6 - is added as an identifier to track of sections:
```

Comment:
what do you mean "to track of sections"?

Reply:
is added as a LAS section identifier